operator/webhook/addons/certmanager/main.py

```diff
@@ -0,0 +1,146 @@
+
+    alt_names = _get_annotation_vals_as_list(
+        annotations.get("cert-manager.io/alt-names")
+    )
+    organizational_units = _get_annotation_vals_as_list(
+        annotations.get("cert-manager.io/subject-organizationalunits")
+    )
+    countries = _get_annotation_vals_as_list(
+        annotations.get("cert-manager.io/subject-countries")
+    )
+    provinces = _get_annotation_vals_as_list(
+        annotations.get("cert-manager.io/subject-provinces")
+    )
+    localities = _get_annotation_vals_as_list(
+        annotations.get("cert-manager.io/subject-localities")
+    )
+
+    dns_names = [
+        f"{common_name}.{namespace}.svc.cluster.local",
+        f"{common_name}.{namespace}.svc",
+        f"{common_name}.{namespace}",
+        common_name,
+        f"{name}-actuator.{namespace}.svc.cluster.local",
+    ]
+    dns_names = dns_names + alt_names
+
+    subject = {}
+    if organizational_units:
+        subject["organizationalUnits"] = organizational_units
+
+    if countries:
+        subject["countries"] = countries
+
+    if provinces:
+        subject["provinces"] = provinces
+
+    if localities:
+        subject["localities"] = localities
+
+    keystore_type = _get_keystore_type(obj)
+    password_secret_ref_name = _get_password_secret_ref_name(obj)
+
+    cert = {
+        "apiVersion": "cert-manager.io/v1",
+        "kind": "Certificate",
+        "metadata": {
+            "name": f"{name}-certs",
+            "namespace": namespace,
+        },
+        "spec": {
+            "commonName": f"{common_name}.{namespace}",
+            "dnsNames": dns_names,
+            "issuerRef": {
+                "group": "cert-manager.io",
+                "kind": issuer_kind,
+                "name": issuer,
+            },
+            "keystores": {
+                keystore_type: {
+                    "create": True,
+                    "passwordSecretRef": {
+                        "key": "password",
+                        "name": password_secret_ref_name,
+                    },
+                }
+            },
+            "secretName": f"{name}-certstore",
+            "subject": subject,
+        },
+    }
+
+    return cert
+
+
+def _get_annotation_vals_as_list(annotation_val) -> List:
+    return (
+        [val for i in annotation_val.split(",") if (val := i.strip())]
+        if annotation_val
+        else []
+    )
+
+
+def _get_keystore_type(obj) -> str:
+    return obj["spec"]["tls"]["keystore"]["type"]
+
+
+def _get_password_secret_ref_name(obj) -> str:
+    return obj["spec"]["tls"]["keystore"]["passwordSecretRef"]
+
+
+def sync_certificate(body) -> Mapping:
+    # Request API at for DecoratorController at https://metacontroller.github.io/metacontroller/api/decoratorcontroller.html#sync-hook-request
+    obj = body["object"]
+    certificate = _new_certificate(obj)
+    attachments = [certificate] if certificate else []
+    desired_state = {"status": {}, "attachments": attachments}

```

Comment:
:exclamation: It's not entirely clear, but from the [Sync Hook Response docs](https://metacontroller.github.io/metacontroller/api/decoratorcontroller.html#sync-hook-response), it seems that setting the `status` field to an empty JSON could potentially clear out the `status` field on the IntegrationRoute, might be best to play it safe and exclude it from the response. We should only set the `status` in the core sync function.

Reply:
Removed status from the certmanager addon response